vscode/src/extension.ts

```diff
@@ -290,13 +296,19 @@ async function updateLanguageServiceEnableFormatting(
   const isFormattingEnabled = getEnableFormating();
   log.debug("Enable formatting set to: " + isFormattingEnabled);
   if (isFormattingEnabled) {
-    formatterHandle.handle =
+    formatterHandle.DocumentFormattingHandle =
       vscode.languages.registerDocumentFormattingEditProvider(
         qsharpLanguageId,
-        createFormatProvider(languageService),
+        createFormattingProvider(languageService),
+      );
+    formatterHandle.DocumentRangeFormattingHandle =
+      vscode.languages.registerDocumentRangeFormattingEditProvider(
+        qsharpLanguageId,
+        createFormattingProvider(languageService),
       );
   } else {
-    formatterHandle.handle?.dispose();
+    formatterHandle.DocumentFormattingHandle?.dispose();
+    formatterHandle.DocumentRangeFormattingHandle?.dispose();
```

Comment:
Minor nit: But best practice would be to set both those values to null after you dispose them.

Reply:
I set them to `undefined` after disposal.